app/src/main/java/sdp/moneyrun/ui/game/GameLobbyActivity.java

```diff
@@ -1,156 +1,211 @@
+
+    private LobbyPlayerListAdapter listAdapter;
+    private Game game;
+    private String gameId;
+    private Player user;
+    private User actualUser;
+    private DatabaseReference thisGame;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_game_lobby);
+
+        addAdapter();
+
+        gameId = getIntent().getStringExtra(getResources().getString(R.string.join_game_lobby_intent_extra_id));
+        user = (Player) getIntent().getSerializableExtra(getResources().getString(R.string.join_game_lobby_intent_extra_user));
+        actualUser = (User) getIntent().getSerializableExtra(getResources().getString(R.string.join_game_lobby_intent_extra_type_user));
+        this.thisGame = FirebaseDatabase.getInstance().getReference()
+                .child(this.getString(R.string.database_games)).child(gameId);
+        getGameFromDb();
+    }
+
+    public LobbyPlayerListAdapter getListAdapter() {
+        return listAdapter;
+    }
+
+    private void addAdapter() {
+        // The adapter lets us add item to a ListView easily.
+        ArrayList<Player> playerList = new ArrayList<>();
+        listAdapter = new LobbyPlayerListAdapter(this, playerList);
+        ListView playerListView = (ListView) findViewById(R.id.lobby_player_list_view);
+        playerListView.setAdapter(listAdapter);
+    }
+
+    /**
+     * @param playerList: players to be added to the leaderboard
+     *                    Adds players to leaderboard
+     */
+    public void addPlayerList(ArrayList<Player> playerList) {
+        if (playerList == null) {
+            throw new NullPointerException("Player list is null");
+        }
+        listAdapter.addAll(playerList);
+    }
+
+    private void getGameFromDb() {
+        GameDatabaseProxy proxyG = new GameDatabaseProxy();
+        proxyG.getGameDataSnapshot(gameId).addOnCompleteListener(task -> {
+            if (task.isSuccessful()) {
+                this.game = proxyG.getGameFromTaskSnapshot(task);
+                setAllFieldsAccordingToGame();
+                listenToIsDeleted();
+                createDeleteOrLeaveButton();
+            } else {
+                Log.e(TAG, task.getException().getMessage());
+            }
+        });
+    }
+
+    private void listenToIsDeleted() {
+        if (!user.equals(game.getHost())) {
+            thisGame.child(DB_IS_DELETED).addValueEventListener(new ValueEventListener() {
+                @Override
+                public void onDataChange(@NonNull DataSnapshot snapshot) {
+                    if ((boolean) snapshot.getValue()) {
+                        Intent intent = new Intent(getApplicationContext(), MenuActivity.class);
+                        intent.putExtra("user", actualUser);
+                        startActivity(intent);
+                        finish();
+                        game.removePlayer(user, false);

```

Comment:
Will this be run even if it is after the finish() function?

Reply:
yes it will